File: Tema 06 - React/petshelter.ui/src/Models/Fundraiser.ts
import type { Person } from "./Person";

export  class Fundraiser {
  
name:	string;
dueDate?: Date; 
goalAmount?:	number;
status?:	string;
donors?:   Person[]; 
currentAmount?:	number;  

constructor(name: string, status?: string,  dueDate?: Date, goalAmount?: number,  donors?: Person[], currentAmount?: number ) {  
    this.name = name; 
    this.dueDate = dueDate; 
    this.goalAmount = goalAmount;
    this.donors = donors;  
    this.status = status; 
    this.currentAmount = currentAmount; 

}
}

File: Tema 07 - Clean Code/PetShelter/PetShelter.Domain/Fundraiser.cs

using Microsoft.EntityFrameworkCore.SqlServer.Query.Internal;

namespace PetShelter.Domain;

public class Fundraiser : INamedEntity
{
        public string Name { get; set; }
        public decimal GoalAmount { get; set; }
    public decimal CurrentAmount { get; set; }

    public DateTime DueDate { get; set; }  
         public string Status { get; set; }  
        public ICollection<Person> Donors { get; set; }
       

    public Fundraiser(string name, decimal goalAmount, DateTime dueDate )
    {  

        Name = name;
        GoalAmount = goalAmount;
        DueDate = dueDate; 
        Donors= new List<Person>();
    }
    
    public Fundraiser(string name, decimal goalAmount, DateTime dueDate, string status, decimal currentAmount)
    { 
        Name = name;
        GoalAmount = goalAmount;
        DueDate = dueDate;
        Status = status;
        CurrentAmount = currentAmount; 
        Donors = new List<Person>();

    }
}
